File: Java/LinkedList/src/domain/Node.java
package domain;

public class Node<T> {
	
	private Node<T> next;
	private T data;
	
	public Node(T _data) {
		this.data = _data;
	}
	
	public T getData() {
		return this.data;
	}
	
	public Node<T> getNext() {
		return this.next;
	}
	
	public void setNext(Node<T> _next) {
		this.next = _next;
	}
}

File: JavaScript/LinkedList/src/Node.js

  function Node(_data) {

    var self = this;
    self.data = _data;
    self.next;

    function getData() {
      return self.data;
    }

    function getNext() {
      return self.next;
    }

    function setNext(_next) {
      self.next = _next;
    }

    return {
      getData: getData,
      getNext: getNext,
      setNext: setNext
    }
  }

  module.exports = Node;
